clocks-impl.cc

```diff
@@ -19,6 +19,21 @@
  * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <seastar/core/print.hh>
+
+#include "db_clock.hh"
+#include "timestamp.hh"
+
 #include "clocks-impl.hh"
 
 std::atomic<int64_t> clocks_offset;
+
+std::ostream& operator<<(std::ostream& os, db_clock::time_point tp) {
+    auto t = db_clock::to_time_t(tp);
+    return os << std::put_time(std::gmtime(&t), "%Y/%m/%d %T");
+}
+
+std::string print_ts(api::timestamp_type ts) {
+    auto t = std::time_t(std::chrono::duration_cast<std::chrono::seconds>(api::timestamp_clock::duration(ts)).count());
+    return format("{}", std::put_time(std::gmtime(&t), "%Y/%m/%d %T"));
+}
```

Comment:
Well, if we look into timestamp.hh, then we see that those timestamps are indeed taken using the clock.
Printing them as Unix time is useful for debugging. I only use `print_ts` in cdc/metadata.cc, in error messages. I could:
1. make `print_ts` a static function in cdc/metadata.cc
2. change `print_ts` so that it prints the integer itself + "as time: (<unix time>)" or something like that

Reply:
The timestamp is a unix timestamp (in microsecond units). Old uses of Cassandra made it completely user defined, but modern Cassandra and Scylla more or less assume it is a unix timestamp. Certainly it's not a problem for debug prints.